docs/DeveloperGuide.md

```diff
@@ -1,38 +1,584 @@
+](https://github.com/AY2122S2-CS2113T-T10-2/tp/blob/master/src/main/java/seedu/planitarium/commands/DeletePersonCommand.java)
+[`AddRecordCommand.java`
+](https://github.com/AY2122S2-CS2113T-T10-2/tp/blob/master/src/main/java/seedu/planitarium/commands/AddRecordCommand.java)
+[`DeleteRecordCommand.java`
+](https://github.com/AY2122S2-CS2113T-T10-2/tp/blob/master/src/main/java/seedu/planitarium/commands/DeleteRecordCommand.java)
+[`EditRecordCommand.java`
+](https://github.com/AY2122S2-CS2113T-T10-2/tp/blob/master/src/main/java/seedu/planitarium/commands/EditRecordCommand.java)
+[`HelpCommand.java`
+](https://github.com/AY2122S2-CS2113T-T10-2/tp/blob/master/src/main/java/seedu/planitarium/commands/HelpCommand.java)
+[`ListCommand.java`
+](https://github.com/AY2122S2-CS2113T-T10-2/tp/blob/master/src/main/java/seedu/planitarium/commands/ListCommand.java)
+[`ListCatCommand.java`
+](https://github.com/AY2122S2-CS2113T-T10-2/tp/blob/master/src/main/java/seedu/planitarium/commands/ListCatCommand.java)
+[`OverviewCommand.java`
+](https://github.com/AY2122S2-CS2113T-T10-2/tp/blob/master/src/main/java/seedu/planitarium/commands/OverviewCommand.java)
+[`SearchCommand.java`
+](https://github.com/AY2122S2-CS2113T-T10-2/tp/blob/master/src/main/java/seedu/planitarium/commands/SearchCommand.java)
+[`ExitCommand.java`
+](https://github.com/AY2122S2-CS2113T-T10-2/tp/blob/master/src/main/java/seedu/planitarium/commands/ExitCommand.java)
+
+The `CommandFactory` provides a `getCommand()` function to get different commands according to the keywords of 
+user's instructions. This keyword is parsed by `Parser`. There're 11 cases, and it would throw an exception called 
+`UnknownInputException` if the keyword from user input is neither of the 11 cases provided. 
+
+The `Command` class is an abstract class that has abstract method ``
+<!-- {For Huilin} -->
+
+### Parser Component
+
+**Class:** [`Parser.java`
+](https://github.com/AY2122S2-CS2113T-T10-2/tp/blob/master/src/main/java/seedu/planitarium/parser/Parser.java)
+
+![ParserClassDiagram](images/ParserClassDiagram.png)
+
+The `Parser` component consists of the `Parser` class, `ParserUtility` class and several `Exception` classes.
+
+The `Parser` class provides the `parseXYZ()` and `getValidXYZ()` methods where `XYZ` is a placeholder for the type of
+term (e.g. `parseKeyword()` and `getValidUserIndex`). The methods prepended by **parse** assists in parsing the user
+input into its respective terms and the method prepended by **getValid** assists in validating the parsed terms and
+returning an appropriately typed object to the `Commands` component. The `Parser` class interacts with the
+`ParserUtility` class which provides supporting methods for parsing and validating. Both classes throws exceptions as
+required.
+
+The following Sequence Diagram shows how the classes of the `Parser` component interacts for each user command.
+
+![ParserOverviewSequenceDiagram](images/ParserSequenceDiagram0.png)
```

Comment:
Perhaps you can consider using reference frames so that it can be broken up into a less complex diagram